# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/src/media-conduit/MediaConduitInterface.h
# Commit: 25bf8e097e60
# Full Hash: 25bf8e097e604cca2842af6754c2c1698db5618a
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2019-10-29 21:27:48
# Description:
#   Bug 1543622 - Make number of channels out param of GetAudioFrame; r=pehrsons
#   
#   The number of channels is available in mAudioFrame in GetAudioFrame so
#   there is no reason to calculate it after the fact in MediaPipeline.
#   
# ==============================================================================

diff -r 4139bfe5e7f9 -r 25bf8e097e60 media/webrtc/signaling/src/media-conduit/MediaConduitInterface.h
--- a/media/webrtc/signaling/src/media-conduit/MediaConduitInterface.h	Tue Oct 29 16:04:29 2019 +0000
+++ b/media/webrtc/signaling/src/media-conduit/MediaConduitInterface.h	Tue Oct 29 17:30:03 2019 +0000
@@ -536,6 +536,8 @@
    *                             Hertz (16000, 32000,..)
    * @param capture_delay [in]: Estimated Time between reading of the samples
    *                            to rendering/playback
+   * @param numChannels [out]: Number of channels in the audio frame,
+   *                           guaranteed to be non-zero.
    * @param lengthSamples [out]: Will contain length of the audio frame in
    *                             samples at return.
    *                             Ex: A value of 160 implies 160 samples each of
@@ -549,7 +551,8 @@
   virtual MediaConduitErrorCode GetAudioFrame(int16_t speechData[],
                                               int32_t samplingFreqHz,
                                               int32_t capture_delay,
-                                              int& lengthSamples) = 0;
+                                              size_t& numChannels,
+                                              size_t& lengthSamples) = 0;
 
   /**
    * Checks if given sampling frequency is supported